query-engine/query-engine-wasm/src/engine.rs

```diff
@@ -125,10 +126,12 @@ impl QueryEngine {
 
         let mut schema = psl::validate(datamodel.into());
         let config = &mut schema.configuration;
+        let preview_features = config.preview_features();
 
         if let Some(adapter) = maybe_adapter {
-            let js_queryable =
-                proxy::from_wasm(adapter).map_err(|e| ApiError::configuration(e.as_string().unwrap_or_default()))?;
+            let js_queryable = driver_adapters::from_wasm(adapter);
+
+            sql_connector::activate_driver_adapter(Arc::new(js_queryable));
```

Comment:
This is a direct adaptation of what we're doing in `query-engine-node-api/src/engine.rs`